app-creator/main.go

```diff
@@ -0,0 +1,145 @@
+package main
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"time"
+	"regexp"
+	"github.com/rs/cors"
+	"modules/handlers"
+	"cloud.google.com/go/datastore"
+	"strings"
+	"github.com/googleapis/google-cloud-go-testing/datastore/dsiface"
+)
+
+func main() {
+	/*
+	  Getting application context.
+	*/
+	ctx := context.Background()
+
+	/* 
+	  Creating a datastore client. This instance is shared across the application.
+	*/
+	db, err := datastore.NewClient(ctx, os.Getenv("GOOGLE_CLOUD_PROJECT"))
+	if err != nil {
+		log.Fatal("Database connection error", err)
+	}
+
+	/*
+	  Create an adapter around the datastore client. The adapter allows us to 
+	  switch between the real datastore client instance and the mock instance for testing.
+	*/
+	dbclient := dsiface.AdaptClient(db);
+
+	/*
+	  Creating shared logger instance.
+	*/
+	l := log.New(os.Stdout, "app-creator", log.LstdFlags)
+
+	/*
+	  Check if a PORT env variable is set.
+	*/
+	port := os.Getenv("PORT")
+
+	/*
+	  Strip any leading colons and make sure it only contains numbers.
+	*/
+	re := regexp.MustCompile(`^:([0-9]+)$`)
+	
+	/*
+	  Default to port 8080 if environment variable is not set or is invalid.
+	*/
+	if port == "" || !re.MatchString(port) {
+		invalidPort := port;
+		port = ":8080"
+		log.Printf("Invalid port number. Received: '%s'. Defaulting to port %s", invalidPort, port)
+	}
+	
+	/*
+	  Initializing a new server mux to handle routes.
+	*/
+	serverMux := http.NewServeMux()
+	
+	/*
+	  Serving static files.
+	*/
+	fs := http.FileServer(http.Dir("./static"))
+	serverMux.Handle("/", fs)
+	serverMux.Handle("/static/", http.StripPrefix("/static/", http.FileServer(http.Dir("./static"))))
+	
+	/*
+	  Setting up handlers.
+	*/
+	handler := handlers.NewHandler(l, dbclient)
+
+	/*
+	  Registering handlers.
+	*/
+	serverMux.Handle("/api/v1/templates", handler)
+
+	/*
+	  CORS middleware. Default option allows all origins. 
+	  https://github.com/rs/cors
+	  ch - cors handler
+	*/ 
+	c := cors.Default().Handler(serverMux)
+
+	var allowedOrigins []string
```

Comment:
I don't think we need this variable at this scope, so please declare when it is initialized.

If you want, the block below can be simplified and de-indented by directly instantiating: `allowedOrigins := strings.Split(os.Getenv("ALLOWED_ORIGINS", ","))` and avoiding `allowedOriginsEnv` altogether. Though, pretty sure you'd still want to check if `len(allowedOrigins) != 0` before creating the CORS handler (not sure what an empty `AllowedOrigins` in `cors.Options` would do).